### PR TITLE
Bugfix/muwm 4184

### PR DESCRIPTION
**Pull request by fanglinfang**
https://jira.cac.washington.edu/browse/MUWM-4184
https://jira.cac.washington.edu/browse/MUWM-4318
built on my-test.

### CODE REVIEW

File: myuw/static/js/card/schedule/visual.js [hunk 1]
```diff
@@ -90,8 +94,20 @@ var VisualScheduleCard = {
...
+
+                    if(meeting.eos_start_date) {
+                        if(!section.has_eos_dates) {
+                            section.has_eos_dates = true;
+                        }
+                    }
+                    meeting.start_end_same = false;
+                    if(meeting.eos_start_date === meeting.eos_end_date) {
+                        meeting.start_end_same = true;
+                    }
```
Comment (ezturner) [inline myuw/static/js/card/schedule/visual.js]: Could this be changed to 
```
meeting.start_end_same = meeting.eos_start_date === meeting.eos_end_date 
```
?

File: myuw/static/js/card/schedule/visual.js [hunk 2]
```diff
@@ -90,8 +94,20 @@ var VisualScheduleCard = {
...
                 var section = this;
+                section.has_eos_dates = false;
                 $.each(section.meetings, function(){
                     var meeting = this;
+
+                    if(meeting.eos_start_date) {
+                        if(!section.has_eos_dates) {
+                            section.has_eos_dates = true;
+                        }
+                    }
```
Comment (ezturner) [inline myuw/static/js/card/schedule/visual.js]: Could this be changed to

```
section.has_eos_dates = meeting.eos_start_date && !section.has_eos_date
```

?

File: myuw/static/js/card/schedule/visual.js [hunk 3]
```diff
@@ -167,7 +183,28 @@ var VisualScheduleCard = {
                         }
                     }
                 });
+
+                if(section.has_eos_dates) {
+                    // sort the meeting by eos_start_date
+                    section.meetings.sort(function(m1, m2) {
```
Comment (ezturner) [inline myuw/static/js/card/schedule/visual.js]: Could this function be refactored out of here and into a file level function or put in a utility file?

File: myuw/static/js/card/schedule/visual.js [hunk 4]
```diff
@@ -167,7 +183,28 @@ var VisualScheduleCard = {
...
+                        }
+                        if(date1 > date2) {
+                            return 1;
+                        }
+                        return 0;
+                    });
+                    visual_data.eos_sections.push(section);
+                    if(!visual_data.has_eos_dates) {
+                        visual_data.has_eos_dates = true;
+                    }
```
Comment (ezturner) [inline myuw/static/js/card/schedule/visual.js]: Could this be:

```
visual_data.has_eos_dates = !visual_data.has_eos_dates
```
?

File: myuw/static/js/card/summary/schedule.js [hunk 5]
```diff
@@ -95,7 +96,7 @@ var SummaryScheduleCard = {
                 case "B-term":
                     b_term.push(instructed_course_data.sections[i]);
                     break;
-                case "Full-term":
+                default:
```
Comment (ezturner) [inline myuw/static/js/card/summary/schedule.js]: What is the rationale for this change? Do PCE/EOS courses not always have a summer term like SDB ones do?

Comment (fanglinfang) [reply myuw/static/js/card/summary/schedule.js]: Yes, this is the fix for MUWM-4318  (EOS summer section has summer term value "".)

File: myuw/static/js/card/schedule/visual.js (hunk 2 above)
Comment (fanglinfang) [reply myuw/static/js/card/schedule/visual.js]: This may set section.has_eos_dates from true to false.

File: myuw/static/js/card/schedule/visual.js (hunk 4 above)
Comment (fanglinfang) [reply myuw/static/js/card/schedule/visual.js]: No, this may change visual_data.has_eos_dates from true to false.
